rt/rs/microprofile-client/src/main/java/org/apache/cxf/microprofile/client/cdi/CDIUtils.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.cxf.microprofile.client.cdi;
+
+import java.util.NoSuchElementException;
+
+import javax.enterprise.context.spi.CreationalContext;
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.BeanManager;
+import javax.enterprise.inject.spi.CDI;
+
+public final class CDIUtils {
+
+    private CDIUtils() {
+    }
+
+    public static <T> T getInstanceFromCDI(Class<T> clazz) {
+        T t;
+        try {
+            t = findBean(clazz);
+        } catch (ExceptionInInitializerError | NoClassDefFoundError | IllegalStateException ex) {
+            // expected if no CDI implementation is available
+            t = null;
+        } catch (NoSuchElementException ex) {
+            // expected if ClientHeadersFactory is not managed by CDI
+            t = null;
+        }
+        return t;
+    }
+
+    @SuppressWarnings("unchecked")
+    private static <T> T findBean(Class<T> clazz) {
+        BeanManager beanManager = CDI.current().getBeanManager();
+        Bean<?> bean = beanManager.getBeans(clazz).iterator().next();
+        CreationalContext<?> ctx = beanManager.createCreationalContext(bean);
```

Comment:
if !beanManager.isNormalScope(bean.getScope()) then ctx must be released to not leak

Reply:
@rmannibucau That makes sense.  When does the ctx need to be released?  Can I release it as soon as I have a reference to the bean, or do I need to wait until it is out of scope?  Thanks for the review!

Reply:
It will trigger dependent predestroys so out of scope I think.

Reply:
I added a new commit that will release the contexts (for all scopes) when the client instance is closed.  Can you take one more peek at my changes?  If all looks good, I think we can merge.  Thanks again!

Reply:
Works since there is one factory per bus
Side note being: why cdi utility does not return an Instance{value,runnable} and you use the runnable as a close callback to call the release method. Main difference is then the storage can be an instance list (not set ;)) in the factory instead of a static method which would leak if anything happens wrong in the app lifecycle.

Reply:
I like it!  I've just made those changes.  One last (hopefully) look?  Thanks again!

Reply:
Second param should be sthg like:

     beanManager.isNormalScope(bean.getScope()) ? () -> {} : context::release

Normal scope lifecycle is already handled by the container.

Otherwise perfect for me.

Reply:
Shouldn't it be the other way around?   We should call release if it _is_ a normal scope, right?

Reply:
If it is not. If it is, it is implicitly called, when not (kind of caller managed) it is your responsability.

Reply:
Those changes are in - as well as updated unit tests.  Thanks again!